sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -345,6 +346,66 @@ case class MakeInterval(
     )
 }
 
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(years, months) - Make year-month interval from years, months.",
+  arguments = """
+    Arguments:
+      * years - the number of years, positive or negative
+      * months - the number of months, positive or negative
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_(1, 2);
+       1-2
+      > SELECT _FUNC_(1, 0);
+       1-0
+      > SELECT _FUNC_(0, 1);
+       0-1
+  """,
+  since = "3.2.0",
+  group = "datetime_funcs")
+// scalastyle:on line.size.limit
+case class MakeYMInterval(years: Expression, months: Expression)
+  extends BinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+
+  def this(years: Expression) = this(years, Literal(0))
+  def this() = this(Literal(0))
+
+  override def left: Expression = years
+  override def right: Expression = months
+  override def inputTypes: Seq[AbstractDataType] = Seq(IntegerType, IntegerType)
+  override def dataType: DataType = YearMonthIntervalType
+  override def nullable: Boolean = children.exists(_.nullable)
+
+  private def evalIntValue(dt: DataType, value: Any): Long = dt match {
+    case _: ByteType | _: ShortType | _: IntegerType => value.asInstanceOf[Number].longValue()
+    case _: LongType => value.asInstanceOf[Long]
+  }
+
+  override def nullSafeEval(year: Any, month: Any): Any = {
+    LongExactNumeric.toInt(Math.addExact(evalIntValue(right.dataType, month),
+      Math.multiplyExact(evalIntValue(left.dataType, year), MONTHS_PER_YEAR)))
+  }
+
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    defineCodeGen(ctx, ev, (years, months) => {
+      val extractor = LongExactNumeric.getClass.getName.stripSuffix("$")
+      s"""
+         |$extractor.toInt(java.lang.Math.addExact($months,
+         |  java.lang.Math.multiplyExact($years, $MONTHS_PER_YEAR)))
+         |""".stripMargin
+    })
+  }
+
+  override def prettyName: String = "make_ym_interval"
+
+  override protected def withNewChildrenInternal(
+      newLeft: Expression,
+      newRight: Expression): Expression =
+    copy(years = newLeft, months = newRight)
+}
```

Comment:
Could you clarify (add comments) why do you need to override `withNewChildrenInternal`.

Reply:
![image](https://user-images.githubusercontent.com/46485123/121983204-847c7d80-cdc3-11eb-8a51-7ee43601e2a5.png)

Reply:
ok. I see. Thanks.